### PR TITLE
Challenge 3

### PR DESCRIPTION
**Pull request by jcastaneyra**
Added features:

- youtube api usage
- api search
- hooks for search, result list and current
- view for detail

image

### CODE REVIEW

File: src/components/VideoDetail/index.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          )
+          .map((item) => (
+            <Styled.VideoScreenshotContainer>
+              <VideoCard
+                onClick={() => showVideoDetail(item)}
+                key={item.id.videoId}
+                title={item.snippet.title}
+                url={item.snippet.thumbnails.medium.url}
+              />
+            </Styled.VideoScreenshotContainer>
```
Comment (jparciga) [inline src/components/VideoDetail/index.jsx]: Remember that the key should go on the root element:

```suggestion
            <Styled.VideoScreenshotContainer key={item.id.videoId}>
              <VideoCard
                onClick={() => showVideoDetail(item)}
                title={item.snippet.title}
                url={item.snippet.thumbnails.medium.url}
              />
            </Styled.VideoScreenshotContainer>
```

**Review by jparciga — COMMENTED** [review]
# Mini-Challenge 3 Feedback:
Good Job, just note that the test cases for the new functionalities are missing.

## Acceptance Criteria
✅  Videos in the Home View are fetched from the YouTube API and displayed correctly.
✅  Users can search for YouTube videos using the search field on the header.
✅  A video can be played from the Video Details View after clicking on it.
✅  The video information and related videos list are displayed correctly on the Video Details View.
✅  When a user clicks on a related video the video player, information and related videos list are updated.

## Bonus Points
❌  Custom Hooks for API calls are implemented and tested correctly.
❌  Testing coverage is above 60%. (Please include a screenshot of the code coverage report in your PR description).

File: src/components/VideoDetail/index.jsx (hunk 1 above)
Comment (jcastaneyra) [reply src/components/VideoDetail/index.jsx]: Hey tocayo @jparciga, yes I know, I am behind on the tests, need to work on that.

Comment (jparciga) [reply src/components/VideoDetail/index.jsx]: No rush, just let me know once you have them 👍🏻

**Pull request merged**